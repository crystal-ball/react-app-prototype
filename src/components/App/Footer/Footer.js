import React from 'react'
import { Flex, Icon, Text } from 'componentry'
import { css } from '@emotion/core'

const footerStyles = ({ colors }) => css`
  padding: 15px 30px;
  background: ${colors.lightGrayBg};

  .icon-heart {
    color: #fcc4dd;
  }
`

const Footer = () => (
  <Flex as='footer' align='center' justify='center' css={footerStyles}>
    <Text mb={0} color='white'>
      <Icon id='heart' />
      Make something awesome!
    </Text>
  </Flex>
)

export default Footer
