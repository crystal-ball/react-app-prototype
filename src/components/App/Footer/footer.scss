@import 'theme';

:local(.component) {
  padding: 15px 30px;
  background: $gray-600;

  .icon-heart {
    color: #fcc4dd;
  }
}
