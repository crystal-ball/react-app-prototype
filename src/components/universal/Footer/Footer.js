import React from 'react'
import { Flex, Icon, Text } from 'componentry'
import { css } from '@emotion/core'

const footerStyles = ({ colors }) => css`
  margin-top: auto;
  padding: 15px 30px;
  background: ${colors.lightGrayBg};

  .icon-heart {
    color: #fcc4dd;
  }
`

const Footer = () => (
  <Flex as='footer' align='center' justify='center' css={footerStyles}>
    <Text mb={0} color='white'>
      <Icon id='heart' data-testid='heart' />
      Make something awesome!
    </Text>
  </Flex>
)

export default Footer
