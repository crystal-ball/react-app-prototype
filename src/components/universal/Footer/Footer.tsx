import { Flex, Icon, Text } from 'componentry'

export function Footer(): JSX.Element {
  return (
    <Flex
      as='footer'
      align='center'
      backgroundColor='ultra'
      borderColor='100'
      justify='center'
      mt='xl'
      px='xl'
      py='md'
      borderTop
    >
      <Icon
        data-testid='heart'
        id='heart'
        fontColor='love'
        mr='xs'
        // Stroke is to provide an example of toHaveStyle
        style={{ stroke: '##a8ffdb' }}
      />
      <Text color='accent' mb={0}>
        Make something awesome!
      </Text>
    </Flex>
  )
}
