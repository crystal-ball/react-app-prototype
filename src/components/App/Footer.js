import React from 'react'
import { Flex, Icon, Text } from 'componentry'

import { component } from './footer.scss'

const Footer = () => (
  <>
    <Flex className={component} align='center' justify='center' p={5}>
      <Text size='lg' mb={0}>
        <Icon id='cog' mr={5} />
        Configuration Generator
      </Text>
    </Flex>
    <Flex as='footer' className='mx-3' align='center' justify='end'>
      <Text textAlign='right' mb={0}>
        <Icon id='warning' className='text-warning' mr={5} /> Warning: This package
        contains awesomeness!
      </Text>
    </Flex>
  </>
)

export default Footer
