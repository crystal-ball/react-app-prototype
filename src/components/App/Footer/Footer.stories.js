/* global module */

import React from 'react'
import { storiesOf } from '@storybook/react'

import Footer from './Footer'

storiesOf('App', module).add('<Footer />', () => <Footer />)
