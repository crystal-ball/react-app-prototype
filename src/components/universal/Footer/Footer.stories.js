/* global module */

import React from 'react'
import { storiesOf } from '@storybook/react'

import Footer from './Footer'
import { emotionWrapper } from '@/theme/emotion'

storiesOf('Universal', module).add('<Footer />', () => emotionWrapper(<Footer />))
