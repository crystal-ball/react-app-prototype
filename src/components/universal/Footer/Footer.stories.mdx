import { Canvas, Meta, Story } from '@storybook/addon-docs/blocks'
import { Footer } from './Footer'

<Meta title='Universal/Footer' component={Footer} />

# Footer

Application footer component contains a fun message.

<Canvas>
  <Story name='<Footer />'>
    <Footer />
  </Story>
</Canvas>
