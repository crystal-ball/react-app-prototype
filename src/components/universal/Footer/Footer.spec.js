import { render, screen } from '@/utils/testing-library'
import { Footer } from './Footer'

describe('<Footer />', () => {
  // ℹ️ example of testing classes, attrs and styles
  it('The icon is an accessible pink font icon', () => {
    render(<Footer />)

    const heartIcon = screen.getByTestId('heart')

    expect(heartIcon).toHaveClass('🅲-icon icon-heart')
    expect(heartIcon).toHaveAttribute('role', 'img')
    expect(heartIcon).toHaveStyle('stroke: ##a8ffdb')
  })

  it('Footer snapshot', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
