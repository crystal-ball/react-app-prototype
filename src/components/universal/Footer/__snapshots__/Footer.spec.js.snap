// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Footer snapshot 1`] = `
<footer
  class="ðŸ…²-flex flex border-top background-color-ultra border-color-100 items-center justify-center mt-xl px-xl py-md"
>
  <svg
    class="ðŸ…²-icon font-variant icon-heart font-color-love mr-xs"
    data-testid="heart"
    role="img"
    style="stroke: ##a8ffdb;"
  >
    <use
      href="#heart"
      xlink:href="#heart"
    />
  </svg>
  <p
    class="ðŸ…²-text body-variant font-color-accent"
    style="margin-bottom: 0px;"
  >
    Make something awesome!
  </p>
</footer>
`;
